ethstorage/downloader/downloader.go

```diff
@@ -220,14 +220,20 @@ func (s *Downloader) downloadToCache() {
 	}
 	s.mu.Unlock()
 
-	// @Qiang devnet-4 have issues to get blob event for the latest block, so if we need roll back to devnet-4
-	// we may need to change it to s.downloadRange(start, end, true)
-	_, err := s.downloadRange(start+1, end, true)
-
-	if err == nil {
-		s.lastCacheBlock = end
-	} else {
-		s.log.Info("DownloadRange failed", "err", err)
+	for start < end {
+		rangeEnd := start + downloadBatchSize
+		if rangeEnd > end {
+			rangeEnd = end
+		}
+		_, err := s.downloadRange(start+1, rangeEnd, true)
+
+		if err == nil {
```

Comment:
Anything should be done to handle the error besides just logging it? For another, would it be more natural to follow the "err!=nil" pattern?

Reply:
yes, err != nil would look better.
if err != nil {
        s.log.Error("DownloadRange failed", "err", err)
	 return
}
s.lastCacheBlock = rangeEnd
start = rangeEnd

Reply:
> Anything should be done to handle the error besides just logging it?
The error is most likely due to network conditions and RPC endpoint errors. We don’t need to retry here because after a slot, another latest block will arrive and call the `downloadToCache` function again.